src/OrchardCore.Modules/OrchardCore.DataProtection.Azure/BlobOptionsSetup.cs

```diff
@@ -11,7 +11,7 @@
 
 namespace OrchardCore.DataProtection.Azure;
 
-public class BlobOptionsSetup
+public class BlobOptionsSetup : IAsyncConfigureOptions<BlobOptions>
```

Comment:
How about IPostConfigureAsync. Reason I suggest post, is because this happens at the end and I think before PostConfigure.

It would be awesome if somehow we have IOptionsAsync that is configured early. But I think having IPostConfigureAsync should be used everywhere we use ISiteService to prevent the blocking call during the bootstrapping of the tenant.

Reply:
It looks like but it is not the same as the `IOptions<>` pattern, would need much more works, for example to support options monitor, snapshot, named options and so on.

But anyway in the `IOptions<>` pattern all `IConfigureOptions<>` are executed (`IPostConfigureOptions` last) when you resolve `IOptions<SomeOptions>`, for example when injected through a constructor, so it can't be async.

Here `ConfigureAsync()` is called just after the tenant container is created, so it will be the first one, before the resolution of any other `IOptions<>`, but here `ConfigureAsync()` should not be used for an option that is also resolved with `IOptions<SomeOptions>`, should only be used with options that will be a singleton being resolved with `SomeOptions`, that's why here we only allow to use this pattern with options being an `IAsyncOptions`.